pkg/cloud/metadata.go

```diff
@@ -75,15 +75,20 @@ func (m *Metadata) GetOutpostArn() arn.ARN {
 func NewMetadata() (MetadataService, error) {
 	sess := session.Must(session.NewSession(&aws.Config{}))
 	svc := ec2metadata.New(sess)
-	// creates the in-cluster config
-	config, err := rest.InClusterConfig()
-	if err != nil && !svc.Available() {
-		return nil, err
-	}
-	// creates the clientset
-	clientset, err := kubernetes.NewForConfig(config)
-	if err != nil && !svc.Available() {
-		return nil, err
+	var clientset *kubernetes.Clientset
+	if !svc.Available() {
+		// creates the in-cluster config
+		config, err := rest.InClusterConfig()
+		if err != nil {
+			return nil, err
+		}
+		// creates the clientset
+		clientset, err = kubernetes.NewForConfig(config)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		clientset = nil
```

Comment:
just curious - do you know why we need to explicitly set clientset to nil here? i thought a pointer's zero-value was nil in Go

Reply:
It's not needed, arguably it's good to be explicit / to test our assumption that it is nil but , as I commented above the issue is because the interface can't be nil